common/tree/type.go

```diff
@@ -4,9 +4,20 @@ import (
 	"github.com/vulpemventures/go-elements/psetv2"
 )
 
+type SigningType uint8
+
+const (
+	// SignAll makes the signer sign all the branches
+	SignAll SigningType = iota
+	// SignBranch makes the signer sign only its own branch
+	SignBranch
+)
+
 type TreeFactory func(outpoint psetv2.InputArgs) (VtxoTree, error)
 
 type VtxoLeaf struct {
-	PubKey string
-	Amount uint64
+	PubKey            string
+	Amount            uint64
+	Type              SigningType
+	SignersPublicKeys []string
```

Comment:
like for the protos, let's put these fields in a `Musig2` struct. Also rename them `Type` > `SigningType` and `SignersPublicKeys` > `CosignerPubkeys`